pg_backup_api/pg_backup_api/logic/utility_controller.py

```diff
@@ -108,30 +111,29 @@ def servers_operation_id_get(server_name: str, operation_id: str) \
     """
     ``GET`` request to ``/servers/*server_name*/operations/*operation_id*``.
 
-    Get status of a recovery operation with ID *operation_id* for Barman server
-    named *server_name*.
+    Get status of an operation with ID *operation_id* for Barman server named
+    *server_name*.
 
-    :param server_name: name of the Barman server related to the recovery
-        operation.
-    :param operation_id: ID of the recovery operation previously created
-        through pg-backup-api.
+    :param server_name: name of the Barman server related to the operation.
+    :param operation_id: ID of the operation previously created through
+        pg-backup-api.
     :return: if *server_name* and *operation_id* are valid, return a JSON
         response containing these keys:
 
-        * ``recovery_id``: the same as *operation_id*;
-        * ``status``: status of recovery process. Maybe be one among: ``DONE``,
+        * ``operation_id``: the same as *operation_id*;
+        * ``status``: status of the operation. Maybe be one among: ``DONE``,
           ``FAILED`` or ``IN_PROGRESS``.
 
         If either *server_name* or *operation_id* is invalid -- or both --
         return a HTTP 400 response with the relevant error message.
     """
     try:
-        operation = ServerOperation(server_name, operation_id)
-        status = operation.get_status_by_operation_id()
-        response = {"recovery_id": operation_id, "status": status}
+        op_server = OperationServer(server_name)
+        status = op_server.get_operation_status(operation_id)
+        response = {"operation_id": operation_id, "status": status}
```

Comment:
@gonzalemario Does this change actually affect repmgr? As far as I can tell repmgr only uses the POST response to this endpoint which is already returning an `operation_id`.

Reply:
No, it doesn't. I thought the same when I first read it. That specific code is indeed changing the response repmgr receives but when we parse it in the callback[1], we just look for the `status` field, not for `recovery_id` (which it's called `operation_id` now). So we almost broke repmgr's standby creation though the API but because the client ignores other fields, we're safe.

In pg-backup-api's  we've got 2 endpoints that receive POST data, this one checks if the previous recovery operation was completed or not. The other endpoint creates the recovery task but it's not used by `servers_operation_id_get()` method.

[1] size_t receive_operation_status(void *content, size_t size, size_t nmemb, char *buffer);